internal/helm/service.go

```diff
@@ -18,333 +18,155 @@ import (
-}
+func validate() error {
+	// name is not empty
+	// name matches a regex
+	// url is a valid URL
 
-// HelmService provides an interface for using Helm on a specific cluster.
-type HelmService struct {
-	clusters ClusterService
+	// external validations:
+	// index is available
+	// password secret exists (if provided)
+	// tls secret exists (if provided)
 
-	logger common.Logger
+	return nil
 }
 
-// NewHelmService returns a new HelmService.
-func NewHelmService(clusters ClusterService, logger common.Logger) *HelmService {
-	return &HelmService{
-		clusters: clusters,
+//go:generate mga gen mockery --name Service --inpkg
+// +kit:endpoint:errorStrategy=service
 
-		logger: logger.WithFields(map[string]interface{}{"component": "helm"}),
-	}
+// Service manages Helm chart repositories.
+type Service interface {
+	// AddRepository adds a new Helm chart repository.
+	AddRepository(ctx context.Context, organizationID uint, repository Repository) error
+
+	// ListRepositories lists Helm repositories.
+	ListRepositories(ctx context.Context, organizationID uint) (repos []Repository, err error)
+
+	DeleteRepository(ctx context.Context, organizationID uint, repoName string) error
+
+	// GetRepositoryIndex(ctx context.Context, organizationID uint, repoName string) (index []byte, err error)
+	// PurgeIndexCache(ctx context.Context, organizationID uint, repoName string) error
 }
 
-// InstallDeployment installs a deployment on a specific cluster.
-// If it's already installed, InstallDeployment does nothing.
-// If it's in a FAILED state, InstallDeployment attempts to delete it first.
-func (s *HelmService) InstallDeployment(
-	ctx context.Context,
-	clusterID uint,
-	namespace string,
-	chartName string,
-	releaseName string,
-	values []byte,
-	chartVersion string,
-	wait bool,
-) error {
-	logger := s.logger.WithContext(ctx).WithFields(map[string]interface{}{"chart": chartName, "release": releaseName})
-	logger.Info("installing deployment")
-
-	cluster, err := s.clusters.GetCluster(ctx, clusterID)
-	if err != nil {
-		return err
-	}
+// NewService returns a new Service.
+func NewService(store Store, secretStore SecretStore, validator RepoValidator, logger common.Logger) Service {
 
-	foundRelease, err := findRelease(releaseName, cluster.KubeConfig)
-	if err != nil {
-		return errors.WithDetails(err, "chart", chartName)
+	return service{
+		store:         store,
+		secretStore:   secretStore,
+		repoValidator: validator,
+		logger:        logger,
 	}
+}
 
-	if foundRelease != nil {
-		switch foundRelease.GetInfo().GetStatus().GetCode() {
-		case release.Status_DEPLOYED:
-			logger.Info("deployment is already installed")
-
-			return nil
-		case release.Status_FAILED:
-			err := helm.DeleteDeployment(releaseName, cluster.KubeConfig)
-			if err != nil {
-				return errors.WrapIfWithDetails(
-					err, "failed to delete deployment",
-					"chart", chartName,
-					"release", releaseName,
-				)
-			}
-		}
-	}
+// +testify:mock:testOnly=true
```

Comment:
```suggestion
// +testify:mock:testOnly=true

```

Empty line ensures that the marker will not be part of the generated documentation. Doesn't make much sense here, still, it's the convention.